src/grt/src/fastroute/src/utility.cpp

```diff
@@ -348,12 +349,24 @@ void FastRouteCore::assignEdge(int netID, int edgeID, bool processDIR)
     if (gridsX[k] == gridsX[k + 1]) {
       min_y = std::min(gridsY[k], gridsY[k + 1]);
       for (l = 0; l < num_layers_; l++) {
-        layer_grid_[l][k] = v_edges_3D_[l][min_y][gridsX[k]].cap - v_edges_3D_[l][min_y][gridsX[k]].usage;
+        // check if the current layer is vertical to match the edge orientation
+        bool is_vertical = ((l % 2) - layer_orientation_) != 0;
```

Comment:
why do we need this check now and not before?

Reply:
I've found this in my tests with congested designs. When a design has congestion, FastRoute was assigning the edge not honoring the layer direction. This check avoids it.